packages/serum/src/market.ts

```diff
@@ -709,7 +707,12 @@ export class Market {
       if (openOrdersAccount) {
         account = openOrdersAccount;
       } else {
-        account = new Account();
+        const marketAddress = this.address.toBase58();
+        account = PublicKey.createWithSeed(
+          ownerAddress,
+          marketAddress.slice(0, 32),
+          this.programId,
+        );
```

Comment:
Now account's type changed from Account to PublicKey, the type script type checker unfortunately tracks it as any, so it doesn't show you that certain calls like `      signers.push(account);` are now invalid.

I double checked the docs for [createWithSeed](https://solana-labs.github.io/solana-web3.js/classes/PublicKey.html#createWithSeed) and it mentions explicitly that the owner of the account will be the program id, which is a change from how these accounts were managed before, so i'm only 80% sure that this will work with the unmodified on-chain program.

Did you test the code with your UI?

Reply:
Yes you are right, I didn't realised that the `programId` was the owner of such an account. Does that mean that the signer won't be able to sign the creation of new orders, settle funds and all? I 100% need to test it out on a UI!

Thanks for taking a look!

Reply:
No all that will work, just concerned that creation of the account might fail

Reply:
yeh i think the fix is just what i commented below

Reply:
sounds good guys, I'll take a stab at this on monday